src/backend/distributed/shared_library_init.c

```diff
@@ -1288,6 +1289,18 @@ RegisterCitusConfigVariables(void)
 		GUC_NO_SHOW_ALL,
 		NULL, NULL, NULL);
 
+	DefineCustomBoolVariable(
+		"citus.enable_replication_origin_session",
+		gettext_noop("Enable replication origin session for avoiding publication of WAL "
```

Comment:
Wondering about the name, one option could be to be very descriptive about the implementation: `citus.use_replication_origin_for_internal_transfers` 

`"Perform internal data transfers like shard splits, moves, and create_distributed_table with a replication origin to be able to filter them out for the purpose of CDC."`

Another option could be `citus.enable_change_data_capture` and such that we can tie some other logic to this?

Reply:
Ok. I have changed this to "citus.enable_change_data_capture"